test/appRelationsDiscovery_test.js

```diff
@@ -0,0 +1,74 @@
+
+describe('appRelationDiscovery', function() {
+  let flightServiceDocs;
+  let correctSlug;
+  let slugOutput;
+  let output;
+  let correctChannelUpdate;
+  let correctChannelQueue;
+
+  before(async function() {
+    flightServiceDocs = getAsyncApiExamples();
+    output = await getRelations(flightServiceDocs);
+    correctSlug = 'mqtt://localhost:1883,mqtt';
+    slugOutput = output.get(correctSlug);
+    correctChannelUpdate = 'flight/update';
+    correctChannelQueue = 'flight/queue';
+  });
+  it('should return correct slug', function() {
+    expect(output).to.have.key(correctSlug);
+  });
+
+  it('should return correct channels', function() {
+    expect(slugOutput).to.have.all.keys(correctChannelUpdate, correctChannelQueue);
+  });
+
+  it('should return correct subscriber data for flight/update channel', async function() {
+    const updateChannelOutput = slugOutput.get(correctChannelUpdate);
+    expect(updateChannelOutput.sub.keys().next().value).to.equal(flightUpdateSubData.keys().next().value);
+    expect(updateChannelOutput.sub.values().next().value).deep.to.equal(flightUpdateSubData.values().next().value);
+  });
+
+  it('should return correct publisher data for flight/update channel', async function() {
```

Comment:
no these tests look cleaner, you just need to get rid of these iterators that I should never suggest to you. Example:

```
  it('should return correct publisher data for flight/update channel', async function() {
    const updateChannelOutput = slugOutput.get(correctChannelUpdate);
    const correctPubOperation = 'Flight Notifier Service';
    expect(updateChannelOutput.pub).to.have.key(correctPubOperation);
    expect(updateChannelOutput.pub.get(correctPubOperation)).to.deep.equal(flightUpdatePubData.get(correctPubOperation));
  });
```

tbh, even ` expect(updateChannelOutput.pub).to.have.key(correctPubOperation);` is probably not needed because the 2nd `expect` already indirectly validates it, cause if key would not be there the 2nd expect fails anyway